runtime-modules/council/src/lib.rs

```diff
@@ -738,7 +790,44 @@ impl<T: Trait> EnsureChecks<T> {
         }
 
         // ensure that previous stake is unlockable (if any present)
-        Self::ensure_previous_stake_unlockable(council_user_id, &account_id)?;
+        Self::ensure_previous_stake_reusable(council_user_id, &account_id)?;
+
+        Ok(account_id)
+    }
+
+    /// Ensures there is no problem in withdrawing already announced candidacy.
+    fn can_withdraw_candidacy(
+        origin: T::Origin,
+        council_user_id: &T::CouncilUserId,
+    ) -> Result<T::AccountId, Error<T>> {
+        // ensure user's membership
+        let account_id = Self::ensure_user_membership(origin, council_user_id)?;
+
+        // escape when no previous candidacy stake is present
+        if !Candidates::<T>::contains_key(council_user_id) {
+            return Err(Error::NotCandidatingNow);
+        }
+
+        let candidate = Candidates::<T>::get(council_user_id);
+
+        // ensure candidacy was announced in on of previous election cycles
```

Comment:
I think this is typo, should be "was not announced" ?

Reply:
Yes that was a typo. Fixed